lib/mobility/plugins/callback_on_write.rb

```diff
@@ -0,0 +1,33 @@
+require 'mobility'
+require 'i18n_sonder/callback_on_write'
+
+module Mobility
+  module Plugins
+    module CallbackOnWrite
+      class << self
+        def apply(attributes, option)
+          return unless option
+          # include the write instance method so that we override the backend's write method
+          attributes.backend_class.include self
+        end
+      end
+
+      def write(locale, value, options = {})
+        return unless should_execute_callback?(value, attribute, locale)
+        
+        super
+        I18nSonder::CallbackOnWrite.trigger.call(model, locale)
+      end
+
+      private
+
+      # Only execute callback if:
+      # 1) the new value being written is different to the already existing value
+      def should_execute_callback?(value, attribute, locale)
+        old_value = model.read_attribute(attribute)
```

Comment:
When getting the old_value, shouldn't we be doing it in the scope of the locale? If the locale is "fr" for example, won't old_value be fetched in english and compare with the value in french?

Reply:
True. Good catch